frame/balances/src/migration.rs

```diff
@@ -69,3 +69,29 @@ impl<T: Config<I>, A: Get<Vec<T::AccountId>>, I: 'static> OnRuntimeUpgrade
 		migrate_v0_to_v1::<T, I>(&A::get())
 	}
 }
+
+pub struct ResetInactive<T, I = ()>(PhantomData<(T, I)>);
+impl<T: Config<I>, I: 'static> OnRuntimeUpgrade for ResetInactive<T, I> {
+	fn on_runtime_upgrade() -> Weight {
+		let onchain_version = Pallet::<T, I>::on_chain_storage_version();
+
+		if onchain_version == 1 {
+			// Remove the old `StorageVersion` type.
+			frame_support::storage::unhashed::kill(&frame_support::storage::storage_prefix(
+				Pallet::<T, I>::name().as_bytes(),
+				"StorageVersion".as_bytes(),
+			));
+
+			InactiveIssuance::<T, I>::kill();
+
+			// Set storage version to `0`.
+			StorageVersion::new(0).put::<Pallet<T, I>>();
```

Comment:
@ggwpez If we're at version 1 now, then it doesn't quite make sense to regress and go back to version 0? Or are we really treating this all as a revert?

Reply:
It is a trick so that the fixed `v0 -> v1` migration can run again.

Reply:
Can we not just write a proper migration for this? And introduce storage Version 2?

Reply:
0 is defined as "not tracking `InactiveIssuance`", 1 is defined as "tracking `InactiveIssuance`". In Polkadot/Kusama mainnet (which are already at version 1), we are migrating back to 0 and thus no longer tracking `InactiveIssuance`, and then migrating back to 1 (and thus tracking it, this time correctly). There are no new alterations to storage formats or indeed state transition semantics, and thus it makes no sense to introduce new `StorageVersion`s which would in all manner be exactly equivalent to the existing `StorageVersion`s. The only thing we are doing is mending state.

Reply:
Yeah, from the PoV of Polkadot/Kusama this makes sense. However, we should write these migrations in a generic way so that other downstream users (e.g. parachain teams) can use them as well. 

There could at least have been some docs on what this migration does and that it should be executed before the other migration.